lib/agent_runner.rb

```diff
@@ -58,9 +58,14 @@ def self.register(agent)
   end
 
   def self.with_connection
-    ActiveRecord::Base.connection_pool.with_connection do
-      yield
-    end
+    begin
+  	  ActiveRecord::Base.connection_pool.with_connection do
+  			  yield
+  	  end
+  	  rescue ActiveRecord::StatementInvalid => e
```

Comment:
My other question is whether we should be resetting all connections, like
you are now, for only the connection of the thread that actually failed.
I'd be concerned that this might break other currently running threads.

On Thursday, November 19, 2015, jn11 <notifications@github.com> wrote:

> In lib/agent_runner.rb
> <https://github.com/cantino/huginn/pull/1154#discussion_r45321011>:
>
> > @@ -58,9 +58,14 @@ def self.register(agent)
> >    end
> >
> >    def self.with_connection
> > -    ActiveRecord::Base.connection_pool.with_connection do
> > -      yield
> > -    end
> > +    begin
> > +  	  ActiveRecord::Base.connection_pool.with_connection do
> > +  			  yield
> > +  	  end
> > +  	  rescue ActiveRecord::StatementInvalid => e
>
> Probably StatementInvalid is too wide and maybe we should catch only
> PG::ConnectionBad
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cantino/huginn/pull/1154/files#r45321011>.
>